src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs

```diff
@@ -319,7 +319,68 @@ public void Visit(TimeValue timeValue)
 
                 public void Visit(UntypedObjectValue untypedObjectValue)
                 {
-                    throw new ArgumentException($"Unable to serialize type {untypedObjectValue.GetType().FullName} to Json format.");
+                    Visit(untypedObjectValue.Impl);
+                }
+
+                private void Visit(IUntypedObject untypedObject, int index = 0)
+                {
+                    FormulaType type = untypedObject.Type;
+
+                    if (index > 40)
+                    {
+                        return;
+                    }
+
+                    if (type is StringType)
+                    {
+                        _writer.WriteStringValue(untypedObject.GetString());
+                    }
+                    else if (type is NumberType || (type is ExternalType et && et.Kind == ExternalTypeKind.UntypedNumber))
+                    {
+                        _writer.WriteNumberValue(untypedObject.GetDouble());
```

Comment:
>.GetDouble() [](http://example.com/codeflow?start=62&length=12)

Will it lose precision if we have a decimal number? For example, `JSON(Decimal(ParseJSON("123456789012345.6789012345678")))` does return `"123456789012345.6789012345678"`. If we were doing JSON on the result of ParseJSON itself, would it truncate it?

Reply:
IUntypedObject doesn't support decimal, by design

Reply:
Is that true? I see:
IUntypedObject.GetDecimal()

Reply:
Done